pkg/util/azureclient/azuresdk/azblob/blobs.go

```diff
@@ -0,0 +1,97 @@
+package azblob
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+
+	"github.com/openshift/installer-aro-wrapper/pkg/util/azureclient"
+)
+
+// BlobsClient is a minimal interface for Azure BlobsClient
+type BlobsClient interface {
+	DownloadStream(ctx context.Context, containerName string, blobName string, o *azblob.DownloadStreamOptions) (azblob.DownloadStreamResponse, error)
+	UploadBuffer(ctx context.Context, containerName string, blobName string, buffer []byte, o *azblob.UploadBufferOptions) (azblob.UploadBufferResponse, error)
+	DeleteBlob(ctx context.Context, containerName string, blobName string, o *azblob.DeleteBlobOptions) (azblob.DeleteBlobResponse, error)
+	BlobExists(ctx context.Context, container string, blobPath string) (bool, error)
+	DeleteContainer(ctx context.Context, container string) error
+}
+
+type blobsClient struct {
+	*azblob.Client
+}
+
+var _ BlobsClient = &blobsClient{}
+
+// NewBlobsClientUsingSAS creates a new BlobsClient using SAS
+func NewBlobsClientUsingSAS(ctx context.Context, sasURL string, environment *azureclient.AROEnvironment) (*blobsClient, error) {
+	options := &azblob.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Cloud: environment.Cloud,
+		},
+	}
+	client, err := azblob.NewClientWithNoCredential(sasURL, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return &blobsClient{
+		Client: client,
+	}, nil
+}
+
+// NewBlobsClientUsingEntra creates a new BlobsClient Microsoft Entra credentials
+func NewBlobsClientUsingEntra(ctx context.Context, environment *azureclient.AROEnvironment, serviceURL string, credential azcore.TokenCredential) (*blobsClient, error) {
+	options := &azblob.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Cloud: environment.Cloud,
+		},
+	}
+	client, err := azblob.NewClient(serviceURL, credential, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return &blobsClient{
+		Client: client,
+	}, nil
+}
+
+func (client *blobsClient) BlobExists(ctx context.Context, container string, blobPath string) (bool, error) {
+	blobRef := client.ServiceClient().NewContainerClient(container).NewBlobClient(blobPath)
+	_, err := blobRef.GetProperties(ctx, nil)
+	if err != nil {
+		if bloberror.HasCode(
+			err,
+			bloberror.BlobNotFound,
+			bloberror.ContainerNotFound,
+			bloberror.ResourceNotFound,
+			bloberror.CannotVerifyCopySource,
+		) {
+			return false, nil
+		} else {
+			return false, err
+		}
+	}
+	return true, nil
+}
+
+func (client *blobsClient) DeleteContainer(ctx context.Context, container string) error {
```

Comment:
Same question as over here: https://github.com/Azure/ARO-RP/pull/3878#discussion_r1795725268

Reply:
Replied on the same thread!